src/main/scala/me/jeffmay/neo4j/client/model/Neo4jResponse.scala

```diff
@@ -0,0 +1,63 @@
+package me.jeffmay.neo4j.client.model
+
+import play.api.libs.json.ops._
+import play.api.libs.json._
+
+// TODO Simplify result types into single result model
+// TODO Document all high-level response models
+sealed trait Neo4jResponse
+object Neo4jResponse {
+
+//  implicit val extractor: TypeKeyExtractor[Neo4jResponse] = Json.extractTypeKey[Neo4jResponse].using({
+//    case open: TxnResponse => Some(open.results)
+//    case other => None
+//  }: Neo4jResponse => Option[Seq[StatementResult]], __ \ "results")
+//
+//  implicit val format: OFormat[Neo4jResponse] = Json.formatAbstract[Neo4jResponse] {
+//    case open: JsArray => OFormat.of[TxnResponse]
+//    case JsNull => OFormat.of[TxnResponse]
+//  }
```

Comment:
Play Json Ops was giving me issues here. I decided to give up and try a more coast-to-coast Json transformation approach, where instead of parsing everything in the case class, I would parse only what I need and transform it directly into the desired return type. I'm not settled on which library will be best for this, but my options include:

- Beef up my usage of Play Json to build transformers:
  https://www.playframework.com/documentation/2.4.x/ScalaJsonTransformers
- Learn some json pattern matching and macros:
  https://github.com/propensive/rapture/blob/dev/doc/json.md

From a performance perspective, I'd be more comfortable with writing transformers, but after glancing at the rapture code base, it looks like he has taken performance very seriously. The only issues I can see would be compile-time cost, which isn't much of an issue once this is a published library (and it is pretty small and self-contained).